src/slurm_plugin/fleet_manager.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+# the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import json
+import logging
+import os
+from configparser import ConfigParser
+from logging.config import fileConfig
+
+from botocore.config import Config
+from common.schedulers.slurm_commands import resume_powering_down_nodes, update_all_partitions
+from slurm_plugin.clustermgtd import ComputeFleetStatus, ComputeFleetStatusManager
+from slurm_plugin.instance_manager import InstanceManager
+from slurm_plugin.slurm_resources import CONFIG_FILE_DIR, PartitionStatus
+
+log = logging.getLogger(__name__)
+
+
+class SlurmFleetManagerConfig:
+    DEFAULTS = {
+        "max_retry": 5,
+        "terminate_max_batch_size": 1000,
+        "proxy": "NONE",
+        "logging_config": os.path.join(
+            os.path.dirname(__file__), "logging", "parallelcluster_fleet_manager_logging.conf"
+        ),
+    }
+
+    def __init__(self, config_file_path):
+        self._get_config(config_file_path)
+
+    def __repr__(self):
+        attrs = ", ".join(["{key}={value}".format(key=key, value=repr(value)) for key, value in self.__dict__.items()])
+        return "{class_name}({attrs})".format(class_name=self.__class__.__name__, attrs=attrs)
+
+    def _get_config(self, config_file_path):
+        """Get fleetmanager configuration."""
+        log.info("Reading %s", config_file_path)
+
+        config = ConfigParser()
+        try:
+            config.read_file(open(config_file_path, "r"))
+        except IOError:
```

Comment:
You could annotate _get_config with `@log_exception(log, "reading cluster manager configuration file", catch_exception=IOError, raise_on_error=True)` as done in clustermgtd